File: Sway3i/src/main/java/com/sway3i/service/FeesService.java
package com.sway3i.service;

import com.sway3i.dto.Fees.Request.FeesRequestDTO;
import com.sway3i.dto.Fees.Response.FeesResponseDTO;
import com.sway3i.entities.Fees;

import java.util.List;

public interface FeesService {
    List<FeesResponseDTO> getAllFees();
    FeesResponseDTO getFeesById(Long id);
    FeesResponseDTO createFees(FeesRequestDTO feesRequest);
    FeesResponseDTO updateFees(Long id, FeesRequestDTO updatedFeesRequest);
    void deleteFees(Long id);
}

File: Sway3i-FrontEnd/src/app/services/Fees/fees.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { FeesRequestDTO } from 'src/app/dto/Fees/requests/fees-request-DTO';
import { FeesResponseDTO } from 'src/app/dto/Fees/responses/fees-response-DTO';

@Injectable({
  providedIn: 'root'
})
export class FeesService {
  private apiUrl = `${environment.api}/fees`;

  constructor(private http: HttpClient) { }

  getAllFees(): Observable<FeesResponseDTO[]> {
    return this.http.get<FeesResponseDTO[]>(this.apiUrl);
  }

  getFeesById(id: number): Observable<FeesResponseDTO> {
    return this.http.get<FeesResponseDTO>(`${this.apiUrl}/${id}`);
  }

  createFees(feesRequest: FeesRequestDTO): Observable<FeesResponseDTO> {
    return this.http.post<FeesResponseDTO>(this.apiUrl, feesRequest);
  }

  updateFees(id: number, updatedFeesRequest: FeesRequestDTO): Observable<FeesResponseDTO> {
    return this.http.put<FeesResponseDTO>(`${this.apiUrl}/${id}`, updatedFeesRequest);
  }

  deleteFees(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
